data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/EventKeyFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.event;
+
+/**
+ * A factory for producing {@link EventKey} objects.
+ *
+ * @since 2.9
+ */
+public interface EventKeyFactory {
+    /**
+     * Creates an {@link EventKey} with given actions.
+     *
+     * @param key The key
+     * @param forActions Actions to support
+     * @return The EventKey
+     * @since 2.9
+     */
+    EventKey createEventKey(String key, EventAction... forActions);
+
+    /**
+     * Creates an {@link EventKey} for the default actions, which are all.
+     *
+     * @param key The key
+     * @return The EventKey
+     * @since 2.9
+     */
+    default EventKey createEventKey(final String key) {
+        return createEventKey(key, EventAction.ALL);
+    }
+
+    /**
+     * An action on an Event.
+     *
+     * @since 2.9
+     */
+    enum EventAction {
```

Comment:
What kind of benefit do we get from this? If I have a key that is for reading and give it ALL, is the downside that the processor could unintentionally write to the key?

Reply:
I wasn't originally planning on adding this. But, I realized that we have different validations for PUT/DELETE than we do for GET. Namely, we don't support an empty string for PUT/DELETE. So the primary advantage is to check that the key is valid for the operation when getting the key, rather than waiting until we process it on an event.